lib/libvfio-user.c

```diff
@@ -939,6 +950,9 @@ exec_command(vfu_ctx_t *vfu_ctx, struct vfio_user_header *hdr, size_t size,
         case VFIO_USER_DEVICE_SET_IRQS:
             ret = handle_device_set_irqs(vfu_ctx, hdr->msg_size, fds, *nr_fds,
                                          cmd_data);
+            if (ret < 0) {
```

Comment:
I don't understand why this is here rather than general

Reply:
> I don't understand why this is here rather than general

It would be wrong to blindly do this. `handle_dma_map_or_unmap` can return `<0` but it might have consumed file descriptors.

Reply:
Sure, but this wouldn't be specific to SET_IRQS either. I think the suggestion I made is much easier.

Reply:
Now obsolete becasue of `consume_fd`.